test/skate_web/channels/detours_channel_test.exs

```diff
@@ -0,0 +1,294 @@
+defmodule SkateWeb.DetoursChannelTest do
+  use SkateWeb.ConnCase
+
+  import Phoenix.ChannelTest
+
+  import Test.Support.Helpers
+  import Skate.Factory
+
+  alias Phoenix.Socket
+  alias SkateWeb.{UserSocket, DetoursChannel}
+
+  setup %{conn: conn} do
+    reassign_env(:skate, :valid_token_fn, fn _socket -> true end)
+
+    %{id: authenticated_user_id} = SkateWeb.AuthManager.Plug.current_resource(conn)
+
+    socket =
+      UserSocket
+      |> socket("", %{})
+      |> Guardian.Phoenix.Socket.put_current_resource(%{id: authenticated_user_id})
+
+    start_supervised({Phoenix.PubSub, name: Skate.PubSub})
+
+    populate_db(conn)
+
+    {:ok, %{conn: conn, socket: socket}}
+  end
+
+  def populate_db(conn) do
```

Comment:
nit(non-blocking): btw I'm running into issues with this pattern in `detours_controller_test` with `populate_db_and_get_user` and would prefer that we try and do these in the tests themselves, but I'll leave that up to you, as we can refactor later.

Reply:
I'd also prefer we don't hardcode Ids, but the whole (insert into db, pull out, update snapshot with id, put back into db) rigamarole is annoying, so no worries. Just a small concern.

Reply:
Mmmm I see I see. Let's talk when you get back. I can see the frustration with this, but want to confirm the better pattern (if there is one) before trying to edit